### ISSUE TITLE
Multiple e2e reboot tests failing consistently with "Node [xxx] failed reboot test."

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#12800]
e.g. kubernetes-e2e-gce-reboot/3412/

Identified problems

Reboot each node by dropping all outbound packets for a while and ensure they function afterwards

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:99
Aug 17 08:16:57.838: Node e2e-reboot-minion-6ovw failed reboot test.

Reboot each node by ordering clean reboot and ensure they function upon restart

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:65
Aug 17 08:17:23.058: Node e2e-reboot-minion-6ovw failed reboot test.

Reboot each node by dropping all inbound packets for a while and ensure they function afterwards

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:91
Aug 17 08:17:48.247: Node e2e-reboot-minion-6ovw failed reboot test.

Reboot each node by ordering unclean reboot and ensure they function upon restart

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:71
Aug 17 08:18:13.443: Node e2e-reboot-minion-6ovw failed reboot test.

Reboot each node by triggering kernel panic and ensure they function upon restart

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:77
Aug 17 08:18:38.648: Node e2e-reboot-minion-6ovw failed reboot test.

Reboot each node by switching off the network interface and ensure they function upon switch on

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:83
Aug 17 08:16:32.647: Node e2e-reboot-minion-6ovw failed reboot test.

**Comment by quinton-hoole** [comment]
This has been happening for at least the past week, which is as far back as our archived history of e2e tests goes.

**Comment by quinton-hoole** [comment]
@marekbiskup do you perhaps have any further info?  I've been on vacation.

**Comment by marekbiskup** [comment]
No, unfortunately not - I got another project, so I haven't worked on Kubernetes for the last month.

**Comment by quinton-hoole** [comment]
Fixed by #12824

**Comment by quinton-hoole** [comment]
These tests are still flaky.  Re-opening to investigate.

**Comment by quinton-hoole** [comment]
There are 6 tests in this suite:

1. Reboot each node by dropping all inbound packets for a while and ensure they function afterwards (Failed 10 times in the last 30 runs. Stability: 66 %)
1. Reboot each node by dropping all outbound packets for a while and ensure they function afterwards (Failed 11 times in the last 30 runs. Stability: 63 %)
1. Reboot each node by ordering clean reboot and ensure they function upon restart (Failed 8 times in the last 30 runs. Stability: 73 %)
1. Reboot each node by ordering unclean reboot and ensure they function upon restart (Failed 8 times in the last 30 runs. Stability: 73 %)
1. Reboot each node by switching off the network interface and ensure they function upon switch on (Failed 9 times in the last 30 runs. Stability: 70 %)
1. Reboot each node by triggering kernel panic and ensure they function upon restart  (Failed 7 times in the last 30 runs. Stability: 76 %)

The failure rates above are misleading, as the common failure mode is that one of the tests fails, and then all subsequent tests fail as a result of the first failure.  Due to the configuration of our test framework, the tests run in a random sequence every time.  Digging into the most recent 10 test runs reveals slightly more useful statistics

**Comment by quinton-hoole** [comment]
(run number, followed by the test numbers above, with P for pass, F for fail):

```
* 4052P: 6P, 1P, 3P, 4P, 2P, 5P (all passed)
* 4051P: (all passed)
* 4050F: 5P, 4P, 6F, ...  (remainder failed)
* 4049P: (all passed) 
* 4048P: (all passed) 
* 4047F: 6P, 4F, ... (all remainder failed) 
* 4046F: 1P, 6P, 3F, ... (all remainder failed) 
* 4045P: (all passed) 
* 4044: 
* 4043: 
* 4042: 
```

**Comment by quinton-hoole** [comment]
Specific reasons for failure:

* 4503
```
Sep  8 16:12:26.320: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-9umz
Sep  8 16:12:26.553: INFO: Waiting up to 2m0s for node e2e-reboot-minion-9umz readiness to be false
Sep  8 16:12:27.320: INFO: Wanted all 5 pods to be running and ready. Result: true. Pods: [elasticsearch-logging-v1-yoagg fluentd-elasticsearch-e2e-reboot-minion-3rc1 kube-registry-proxy-e2e-reboot-minion-3rc1 kube-registry-v0-52afi monitoring-heapster-v8-fq43d]
Sep  8 16:12:27.321: INFO: Getting external IP address for e2e-reboot-minion-3rc1
Sep  8 16:12:27.321: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-3rc1
Sep  8 16:12:27.510: INFO: Waiting up to 2m0s for node e2e-reboot-minion-3rc1 readiness to be false
Sep  8 16:12:27.520: INFO: Node e2e-reboot-minion-9umz condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:27.720: INFO: Node e2e-reboot-minion-3rc1 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:30.325: INFO: Waiting for pod kibana-logging-v1-9wmud                                 in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (5.658889962s elapsed)
Sep  8 16:12:32.552: INFO: Node e2e-reboot-minion-9umz condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:32.723: INFO: Node e2e-reboot-minion-3rc1 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:35.329: INFO: Waiting for pod kibana-logging-v1-9wmud                                 in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (10.663636031s elapsed)
Sep  8 16:12:37.556: INFO: Node e2e-reboot-minion-9umz condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:37.727: INFO: Node e2e-reboot-minion-3rc1 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:40.334: INFO: Waiting for pod kibana-logging-v1-9wmud                                 in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (15.668434867s elapsed)
Sep  8 16:12:42.560: INFO: Node e2e-reboot-minion-9umz condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:42.741: INFO: Node e2e-reboot-minion-3rc1 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 16:12:10 -0700 PDT
Sep  8 16:12:45.334: INFO: Pod monitoring-influx-grafana-v1-ya466                      failed to be running and ready.
Sep  8 16:12:45.334: INFO: Wanted all 4 pods to be running and ready. Result: false. Pods: [fluentd-elasticsearch-e2e-reboot-minion-oi4a kibana-logging-v1-9wmud kube-registry-proxy-e2e-reboot-minion-oi4a monitoring-influx-grafana-v1-ya466]
```

**Comment by quinton-hoole** [comment]
* 4500

```
Sep  8 14:56:31.109: INFO: Wanted all 5 pods to be running and ready. Result: true. Pods: [elasticsearch-logging-v1-hv0be fluentd-elasticsearch-e2e-reboot-minion-123n kibana-logging-v1-kbyum kube-registry-proxy-e2e-reboot-minion-123n monitoring-heapster-v8-a5mwz]
Sep  8 14:56:31.109: INFO: Getting external IP address for e2e-reboot-minion-123n
Sep  8 14:56:31.109: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-123n
Sep  8 14:56:31.709: INFO: Waiting for pod monitoring-influx-grafana-v1-tndda                      in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (1.850919955s elapsed)
Sep  8 14:56:31.908: INFO: Waiting for pod elasticsearch-logging-v1-m8zci                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (2.049798254s elapsed)
Sep  8 14:56:32.368: INFO: Waiting up to 2m0s for node e2e-reboot-minion-123n readiness to be false
Sep  8 14:56:32.707: INFO: Node e2e-reboot-minion-123n condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:36.744: INFO: Waiting for pod monitoring-influx-grafana-v1-tndda                      in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (6.886316686s elapsed)
Sep  8 14:56:36.913: INFO: Waiting for pod elasticsearch-logging-v1-m8zci                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (7.054719264s elapsed)
Sep  8 14:56:37.743: INFO: Node e2e-reboot-minion-123n condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:41.750: INFO: Wanted all 3 pods to be running and ready. Result: true. Pods: [fluentd-elasticsearch-e2e-reboot-minion-9r5i kube-registry-proxy-e2e-reboot-minion-9r5i monitoring-influx-grafana-v1-tndda]
Sep  8 14:56:41.750: INFO: Getting external IP address for e2e-reboot-minion-9r5i
Sep  8 14:56:41.750: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-9r5i
Sep  8 14:56:41.917: INFO: Waiting for pod elasticsearch-logging-v1-m8zci                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (12.059428073s elapsed)
Sep  8 14:56:42.345: INFO: Waiting up to 2m0s for node e2e-reboot-minion-9r5i readiness to be false
Sep  8 14:56:42.355: INFO: Node e2e-reboot-minion-9r5i condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:42.747: INFO: Node e2e-reboot-minion-123n condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:46.922: INFO: Waiting for pod elasticsearch-logging-v1-m8zci                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (17.064026638s elapsed)
Sep  8 14:56:47.359: INFO: Node e2e-reboot-minion-9r5i condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:47.751: INFO: Node e2e-reboot-minion-123n condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 14:56:21 -0700 PDT
Sep  8 14:56:51.922: INFO: Pod kube-registry-v0-f4o71                                  failed to be running and ready.
Sep  8 14:56:51.922: INFO: Wanted all 4 pods to be running and ready. Result: false. Pods: [elasticsearch-logging-v1-m8zci fluentd-elasticsearch-e2e-reboot-minion-n58z kube-registry-proxy-e2e-reboot-minion-n58z kube-registry-v0-f4o71]
```

**Comment by quinton-hoole** [comment]
* 4481

```
Sep  8 06:35:18.802: INFO: Waiting for pod elasticsearch-logging-v1-r7mvu                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (802.764816ms elapsed)
Sep  8 06:35:19.298: INFO: Waiting up to 2m0s for node e2e-reboot-minion-gr5e readiness to be false
Sep  8 06:35:19.403: INFO: Waiting for pod monitoring-influx-grafana-v1-utiue                      in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (1.402238856s elapsed)
Sep  8 06:35:19.800: INFO: Node e2e-reboot-minion-gr5e condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:23.807: INFO: Waiting for pod elasticsearch-logging-v1-r7mvu                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (5.807928352s elapsed)
Sep  8 06:35:24.409: INFO: Waiting for pod monitoring-influx-grafana-v1-utiue                      in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (6.408780665s elapsed)
Sep  8 06:35:24.804: INFO: Node e2e-reboot-minion-gr5e condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:28.812: INFO: Waiting for pod elasticsearch-logging-v1-r7mvu                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (10.812789067s elapsed)
Sep  8 06:35:29.415: INFO: Wanted all 3 pods to be running and ready. Result: true. Pods: [kube-registry-proxy-e2e-reboot-minion-5lw3 monitoring-influx-grafana-v1-utiue fluentd-elasticsearch-e2e-reboot-minion-5lw3]
Sep  8 06:35:29.416: INFO: Getting external IP address for e2e-reboot-minion-5lw3
Sep  8 06:35:29.416: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-5lw3
Sep  8 06:35:29.755: INFO: Waiting up to 2m0s for node e2e-reboot-minion-5lw3 readiness to be false
Sep  8 06:35:29.758: INFO: Node e2e-reboot-minion-5lw3 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:29.808: INFO: Node e2e-reboot-minion-gr5e condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:33.817: INFO: Waiting for pod elasticsearch-logging-v1-r7mvu                          in namespace 'kube-system' status to be 'running and ready'(found phase: "Running", readiness: false) (15.817818296s elapsed)
Sep  8 06:35:34.762: INFO: Node e2e-reboot-minion-5lw3 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:34.811: INFO: Node e2e-reboot-minion-gr5e condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 06:35:07 -0700 PDT
Sep  8 06:35:38.817: INFO: Pod fluentd-elasticsearch-e2e-reboot-minion-g6vi            failed to be running and ready.
Sep  8 06:35:38.817: INFO: Wanted all 4 pods to be running and ready. Result: false. Pods: [kube-registry-proxy-e2e-reboot-minion-g6vi kube-registry-v0-77a81 elasticsearch-logging-v1-r7mvu fluentd-elasticsearch-e2e-reboot-minion-g6vi]
```

**Comment by quinton-hoole** [comment]
* 4479

```
Sep  8 05:58:16.016: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down,network not configured correctly", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:21.020: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down,network not configured correctly", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:26.025: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down,network not configured correctly", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:31.029: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:36.033: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:41.037: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:46.042: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:51.046: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:58:56.050: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:59:01.054: INFO: Node e2e-reboot-minion-k3qc condition 1/1: type: Ready, status: False, reason: "container runtime is down,network not configured correctly", message: "", last transition time: 2015-09-08 05:54:02 -0700 PDT
Sep  8 05:59:06.054: INFO: Node e2e-reboot-minion-k3qc didn't reach desired readiness (true) within 5m0s
```

**Comment by quinton-hoole** [comment]
* 4478

```
Sep  8 05:28:38.626: INFO: Calling nohup sh -c 'sleep 10 && echo c | sudo tee /proc/sysrq-trigger' >/dev/null 2>&1 & on e2e-reboot-minion-rrzg
Sep  8 05:28:38.825: INFO: Node e2e-reboot-minion-xpe8 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:38.871: INFO: Waiting up to 2m0s for node e2e-reboot-minion-rrzg readiness to be false
Sep  8 05:28:39.025: INFO: Node e2e-reboot-minion-rrzg condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:41.745: INFO: Node e2e-reboot-minion-rxir condition 1/1: type: Ready, status: False, reason: "network not configured correctly", message: "", last transition time: 2015-09-08 05:28:33 -0700 PDT
Sep  8 05:28:43.829: INFO: Node e2e-reboot-minion-xpe8 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:44.029: INFO: Node e2e-reboot-minion-rrzg condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:46.751: INFO: Node e2e-reboot-minion-rxir condition 1/1: type: Ready, status: False, reason: "network not configured correctly", message: "", last transition time: 2015-09-08 05:28:33 -0700 PDT
Sep  8 05:28:48.833: INFO: Node e2e-reboot-minion-xpe8 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:49.032: INFO: Node e2e-reboot-minion-rrzg condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:51.756: INFO: Node e2e-reboot-minion-rxir condition 1/1: type: Ready, status: False, reason: "network not configured correctly", message: "", last transition time: 2015-09-08 05:28:33 -0700 PDT
Sep  8 05:28:53.837: INFO: Node e2e-reboot-minion-xpe8 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:54.037: INFO: Node e2e-reboot-minion-rrzg condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transition time: 2015-09-08 05:28:06 -0700 PDT
Sep  8 05:28:56.756: INFO: Node e2e-reboot-minion-rxir didn't reach desired readiness (true) within 20s

```

**Comment by quinton-hoole** [comment]
So in 3/5 cases, all nodes became ready but one or more pods did not, and in 2/5 cases at least one nodes never became ready.

**Comment by quinton-hoole** [comment]
It looks like #13866 (reduce pod eviction timeout) adversely affected the failure rate of these tests (not unexpectedly).  The reboot tests need to be tuned correctly to the new pod eviction timeout.

**Comment by quinton-hoole** [comment]
Fixed by  #14784
